packages/overlay/src/integrations/sentry/data/sentryStore.ts

```diff
@@ -410,6 +412,60 @@ const useSentryStore = create<SentryStoreState & SentryStoreActions>()((set, get
     );
   },
 
+  getFunctionProfiles(): FunctionProfile[] {
+    // separate profiles
+    const allProfiles: FunctionProfile[] = [];
+
+    for (const [traceId, profile] of this.profilesByTraceId) {
+      const functionProfiles = new Map<string, FunctionProfile>();
+
+      // Go over each profile sample
+      for (let sampleIdx = 0; sampleIdx < profile.samples.length - 1; sampleIdx++) {
+        const sample = profile.samples[sampleIdx];
+        const nextSample = profile.samples[sampleIdx + 1];
+        const duration = nextSample.start_timestamp - sample.start_timestamp;
+
+        const stackId = sample.stack_id;
+        const frameIndices = profile.stacks[stackId];
+        if (!frameIndices?.length) continue;
+
+        // go over each stack frame (frame = 1 func/meth call)
+        for (const frameIdx of frameIndices) {
+          const frame = profile.frames[frameIdx];
+          if (!frame) continue;
+
+          const funcName =
+            frame.function ||
+            (frame.module
+              ? `${frame.module}:<anonymous>`
+              : frame.filename
+                ? `${frame.filename}:${frame.lineno || '?'}`
+                : '<unknown>');
+
+          const existing = functionProfiles.get(funcName) || {
+            name: funcName,
+            totalTime: 0,
+            samples: 0,
+            frames: [],
+            traceId, //
```

Comment:
We gotta revisit this as a function may be present in multiple traces. This also requires some updates to the UI